auditlog/models/rule.py

```diff
@@ -261,42 +292,79 @@ def read(self, *args, **kwargs):
                 rule_model = self.env['auditlog.rule']
                 rule_model.sudo().create_logs(
                     self.env.uid, self._name, self.ids,
-                    'read', read_values)
+                    'read', read_values, None, {'log_type': log_type})
             return result
         return read
 
+    @api.multi
     def _make_write(self):
         """Instanciate a write method that log its calls."""
+        self.ensure_one()
+        log_type = self.log_type
+
         @api.multi
-        def write(self, vals, **kwargs):
+        def write_full(self, vals, **kwargs):
             self = self.with_context(auditlog_disabled=True)
             rule_model = self.env['auditlog.rule']
             old_values = dict(
                 (d['id'], d) for d in self.sudo()
                 .with_context(prefetch_fields=False).read(list(self._fields)))
-            result = write.origin(self, vals, **kwargs)
+            result = write_full.origin(self, vals, **kwargs)
             new_values = dict(
                 (d['id'], d) for d in self.sudo()
                 .with_context(prefetch_fields=False).read(list(self._fields)))
             rule_model.sudo().create_logs(
                 self.env.uid, self._name, self.ids,
-                'write', old_values, new_values)
+                'write', old_values, new_values, {'log_type': log_type})
+            return result
+
+        @api.multi
+        def write_fast(self, vals, **kwargs):
+            self = self.with_context(auditlog_disabled=True)
+            rule_model = self.env['auditlog.rule']
+            # Log the user input only, no matter if the `vals` is updated
+            # afterwards as it could not represent the real state
+            # of the data in the database
+            vals2 = dict(vals)
+            old_vals2 = dict.fromkeys(vals2.keys(), False)
+            old_values = dict((id_, old_vals2) for id_ in self.ids)
```

Comment:
Yes we need this dictionary with empty values to generate a result when we call the `diff.changed()` in the `create_logs()` method. Another way would be to pass `None` instead of the `old_values` empty dict and to write `diff.changed() or diff.added()` to be compatible with both *Full* and *Fast* logging, but it was not clear enough for me to do that way, and I would like to avoid possible *side effects* with such expression.